### ISSUE TITLE
GWT 2.8.0 Branch Needed - Specific issue: setPrivate method unavailable

### ISSUE THREAD

**Issue by mrkeathley** [Atmosphere/atmosphere-extensions#227]
Compiling on GWT 2.8.0-beta1 was failing due to the setPrivate method not being found as called in the SerializerGenerator class.

I was able to fork and get a working version with this minor (but deprecating for 2.7.x) commit --
https://github.com/mrkeathley/atmosphere-extensions/commit/c4d5eb2b38b1c113ad1a5a8bff4d50dd1a657257

Can we get a GWT 2.8 support branch created so we have a place to pull these fixes into?

**Comment by elakito** [comment]
@mrkeathley we can pull in this particular change without creating a separate branch, though.
setPrivate has been deprecated since 2.2.0 and setVisibility has been available since then.

do we still need a dedicated branch for 2.8.0?

**Comment by mrkeathley** [comment]
@elakito You are correct. It appears the support for 2.8.0 is rather minimal. I think merging in this single change is sufficient.

**Comment by olivergg** [comment]
So, now that GWT 2.8 final is out, will it be merged soon ?

**Comment by elakito** [comment]
I pushed a modified change of mrkeathley (i.e., the constructor change is modified so that it should work for both 2.5 and 2.8) to atmosphere-extension master and 2.4.x. Please test it. If the problem is resolved and there is no regression, we can release 2.4.x.

**Comment by nkuzman** [comment]
SerializerGenerator in version 2.4.x seems to have the same issue with GWT versions 2.7 and 2.8. Compile fails with the known exception:

Caused by: java.lang.NoSuchMethodError: com.google.gwt.user.rebind.rpc.SerializableTypeOracleBuilder.<init>(Lcom/google/gwt/core/ext/TreeLogger;Lcom/google/gwt/core/ext/PropertyOracle;Lcom/google/gwt/core/ext/GeneratorContext;)V
	at org.atmosphere.gwt20.rebind.SerializerGenerator.generateIncrementally(SerializerGenerator.java:87)

**Comment by olivergg** [comment]
@elakito It works fine with 778e17aa2fddf65241b2d942268c39440a5add35. 
@nkuzman  I guess it's not merged on the 2.4.x branch yet, you'll have to build the 2.5.0-SNAPSHOT to test it.

**Comment by olivergg** [comment]
I may have spoken too soon. Even though everything compiles and runs fine, there is now a new issue with the classloader and/or serialization (but it might be related to how GWT 2.8 serializes things now)

Basically, a message with a payload of type T (which implements Serializable) is pushed from the client to the server, I can even inspect its content with the eclipse debbugger on the other side (so the serialization is fine, at least for the content).

Here is the rpcserializer used on the client side :
```java
@GwtRpcSerialTypes({ Serializable.class })
abstract public class RPCSerializer extends GwtRpcClientSerializer
```

And on the server side :
```java
@Post
public void onPostDispatcher(final AtmosphereResource atmosphereResource)
{
final Object message = atmosphereResource.getRequest().getAttribute(Constants.MESSAGE_OBJECT);
// inspecting the message Object with the debugger seems to indicate that it has the proper expected type T.
// but then ...
message instanceof T; // ... returns false instead of true, as it did before
```
If I inspect the classloader of the message class, it is different from the one retrieved from Class<T>, so it might explain why it does that...
The message type has this classloader : `sun.misc.Launcher$AppClassLoader` instead of `WebAppClassLoader`

Should I open a new issue for that ?

**Comment by olivergg** [comment]
@elakito  Regarding the classloader issue, this was only caused by an old jetty version with GWT 2.8. Migrating to tomcat solved all the issues. So I think it's safe to assume that https://github.com/Atmosphere/atmosphere-extensions/commit/778e17aa2fddf65241b2d942268c39440a5add35 could and should be merged to the stable branch. Right now, I had to clone the 2.5.0-SNAPSHOT branch to use this patch.

**Comment by jfarcand** [comment]
@olivergg pull request and I will cut the release of the branch you need

**Comment by olivergg** [comment]
@jfarcand  thanks for your answer and by the way thanks for your amazing work on the Atmosphere framework.
About the pull request, I think the commit is already merged into master.  Actually it's already on the 2.4.x branch.... (according to https://github.com/Atmosphere/atmosphere-extensions/commits/extensions-2.4.x)
Is it enough to publish a new artifact on maven central ?

**Comment by jfarcand** [comment]
@olivergg Released 2.4.4. Should be soon available
